### ISSUE TITLE
Consider using `--enable-install-libiberty` when building gcc or binutils

### ISSUE THREAD

**Issue by neodyne** [skeeto/w64devkit#187]
Mingw-w64 in w64devkit, in contrast to niXman's old, but still fairly popular [mingw builds](https://sourceforge.net/projects/mingw-w64/files/Toolchains%20targetting%20Win64/Personal%20Builds/mingw-builds/8.1.0/)[1] doesn't include [libiberty](https://gcc.gnu.org/onlinedocs/libiberty/).

Could it be included in the distribution by passing `--enable-install-libiberty` when building either gcc or binutils[2], in the Dockerfile?

Usecase is building old 3rd party source code, OOTB.

---
[1] via binutils; binutils<2.24 apparently used to install libiberty by default.
[2] binutils however may or may not [ignore the switch](https://wiki.linuxfromscratch.org/lfs/ticket/3456) altogether.

**Comment by skeeto** [comment]
Thanks for reaching out. This is an interesting idea, but, sorry, I'm 
uninterested in distributing libiberty. In general the only libraries I 
intend to distribute in w64dk are language runtimes (libstdc++, etc.) and 
import libraries for system DLLs (msvcrt, kernel32, etc.).

Besides that, libiberty is, to be frank, quite poorly written. I wouldn't 
feel comfortable encouraging its use. Examining the copy in the GCC tree, 
I didn't see a full page of code that didn't contain a least two obvious 
bugs (undetected integer overflows, invalid pointer use, etc.).

If this is something you really want, it's easy to build a custom w64dk 
for yourself with this feature enabled. However, when I tried it with both 
GCC and Bintuils (in the cross-build, not the bootstrap) it didn't work. I 
got libiberty.h but no library. I suspect this feature is unsupported for 
cross builds.

**Comment by neodyne** [comment]
I understand, it's your decision after all, it was just a suggestion.

>Besides that, libiberty is, to be frank, quite poorly written. I wouldn't
feel comfortable encouraging its use. Examining the copy in the GCC tree,
I didn't see a full page of code that didn't contain a least two obvious
bugs (undetected integer overflows, invalid pointer use, etc.).

I'm not familiar with the source code to a degree, where I could form a qualified opinion, but superficial searching suggests that at least some(?) bugs [got caught](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=88783) in the past/recently.
Although generally, it is a decomissioned package; as said, in my case linking against it is needed for legacy code.

>However, when I tried it with both
GCC and Bintuils (in the cross-build, not the bootstrap) it didn't work. I
got libiberty.h but no library. I suspect this feature is unsupported for
cross builds.

For completeness' sake, I checked both WinLibs and niXman's [current mingw builds](https://github.com/niXman/mingw-builds-binaries), all contain `libiberty.a`, via binutils. The used [build script for binutils]( https://github.com/niXman/mingw-builds/blob/develop/scripts/binutils.sh) however, does quite a bit more than the build step for binutils in w64devkit; there are multiple [patches](https://github.com/niXman/mingw-builds/tree/develop/patches/binutils), ~3 related to libiberty directly.

I'll stick to those prebuilt mingw-w64 toolchains for this usecase.

Thank you for your reply, and for w64devkit. (and your blog - nullprogram.com - I found it very informative!)
